mllib/src/main/scala/org/apache/spark/ml/evaluation/MultilabelClassificationEvaluator.scala

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.evaluation
+
+import org.apache.spark.annotation.{Experimental, Since}
+import org.apache.spark.ml.param._
+import org.apache.spark.ml.param.shared._
+import org.apache.spark.ml.util._
+import org.apache.spark.mllib.evaluation.MultilabelMetrics
+import org.apache.spark.sql.Dataset
+import org.apache.spark.sql.functions._
+import org.apache.spark.sql.types._
+
+
+/**
+ * :: Experimental ::
+ * Evaluator for regression, which expects two input columns: predictions and labels.
+ */
+@Since("3.0.0")
+@Experimental
+class MultilabelClassificationEvaluator @Since("3.0.0") (@Since("3.0.0") override val uid: String)
+  extends Evaluator with HasPredictionCol with HasLabelCol
+    with DefaultParamsWritable {
+
+  @Since("3.0.0")
+  def this() = this(Identifiable.randomUID("mlcEval"))
+
+  /**
+   * param for metric name in evaluation (supports `"f1"` (default), `"weightedPrecision"`,
+   * `"weightedRecall"`, `"accuracy"`)
+   * @group param
+   */
+  @Since("3.0.0")
+  final val metricName: Param[String] = {
+    val allowedParams = ParamValidators.inArray(Array("subsetAccuracy", "accuracy",
+      "hammingLoss", "precision", "recall", "f1Measure", "precisionByLabel",
```

Comment:
You could probably declare this array as a private val in the object, and then reuse it to create the string below too: `metrics.mkString("(", "|", ")")`